tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -397,6 +397,10 @@ export class WebGPUBackend extends KernelBackend {
     this.maybeReleaseBuffer(dataId);
 
     info.values = data;
+    // converted bool to Uint8Array.
+    if (info.dtype === 'bool') {
+      return Uint8Array.from(data);
```

Comment:
Please do it in `ArrayBufferToTypedArray` and keep this file unchanged. And I am curious why an extra `Uint8Array.from` is needed since in `ArrayBufferToTypedArray`, you already return a Uint8Array and the data is assigned one by one there.

Reply:
`ArrayBufferToTypedArray` is not reached when tensorBufferInfo.values is not null, `read` function will return values through `convertAndCacheOnCPU`. For bool tensor, it stores as Int32Array in tensorMap(converted by `write` function), so convert it back to Uint8Array here.

Reply:
If the values are always in CPU, maybe we shouldn't change it to int32Array in `write` https://github.com/tensorflow/tfjs/blob/master/tfjs-backend-webgpu/src/backend_webgpu.ts#L269. And lazily do it in `uploadToGPU`. Then we don't need to do special process here.

Reply:
> `If the values are always in CPU, maybe we shouldn't change it to int32Array in write https://github.com/tensorflow/tfjs/blob/master/tfjs-backend-webgpu/src/backend_webgpu.ts#L269. And lazily do it in uploadToGPU. Then we don't need to do special process here.`

Agree. I removed `Uint8Array.from` from `write`, as uploadToGPU calls TypedArray.prototype.set() to store values, the two arrays may share the same underlying ArrayBuffer.